File: exampleProjects/PythonExample/src/de/sots/cellarsandcentaurs/application/creature_facade.py
import uuid
import de.sots.cellarsandcentaurs.domain.models as models
from de.sots.cellarsandcentaurs.domain.services.creature_service import CreatureService


class CreatureFacade:
    STANDARD_CREATURE_TYPE = models.CreatureType.MONSTROSITY

    def __init__(self, creature_service: CreatureService):
        self.creature_service = creature_service

    def create(self, type: models.CreatureType, walk_speed: models.Speed, fly_speed: models.Speed,
               swim_speed: models.Speed, burrow_speed: models.Speed, climb_speed: models.Speed,
               armor_class: models.ArmorClass, hit_points_value: int):
        creature_id = models.CreatureId(uuid.uuid4())
        creature = models.Creature(creature_id)

        # Set properties using the assumed methods
        creature.armor_class = armor_class
        creature.hit_points = models.HitPoints.init(hit_points_value)
        creature.type = type
        creature.speeds = {
            models.SpeedType.WALKING: walk_speed,
            models.SpeedType.FLYING: fly_speed,
            models.SpeedType.SWIMMING: swim_speed,
            models.SpeedType.BURROWING: burrow_speed,
            models.SpeedType.CLIMBING: climb_speed
        }

        self.creature_service.save(creature)

File: exampleProjects/PhpExample/src/de/sots/cellarsandcentaurs/application/CreatureFacade.php
<?php
namespace de\sots\cellarsandcentaurs\application;

require '../domain/model/Creature.php';

use de\sots\cellarsandcentaurs\domain\model\{ArmorClass,
    Creature,
    CreatureId,
    CreatureType,
    HitPoints,
    Speed,
    SpeedType};
use de\sots\cellarsandcentaurs\domain\service\CreatureService;


class CreatureFacade
{
    const STANDARD_CREATURE_TYPE = CreatureType::MONSTROSITY;
    private CreatureService $creatureService;

    function __construct(CreatureService $creatureService)
    {
       $this->$creatureService = $creatureService;
    }

    public function create(
        CreatureType $type,
        Speed $walkSpeed,
        Speed $flySpeed,
        Speed $swimSpeed,
        Speed $burrowSpeed,
        Speed $climbSpeed,
        ArmorClass $armorClass,
        int $hitPointsValue
    ) : Creature
    {
        $creature = new Creature(new CreatureId($this->generateUUID()));
        $creature->setArmorClass($armorClass);
        $creature->setHitPoints(HitPoints::init($hitPointsValue));
        $creature->setType($type);
        $creature->setSpeeds([
            SpeedType::WALKING->value => $walkSpeed,
            SpeedType::FLYING->value => $flySpeed,
            SpeedType::SWIMMING->value => $swimSpeed,
            SpeedType::BURROWING->value => $burrowSpeed,
            SpeedType::CLIMBING->value => $climbSpeed
        ]);

        $this->creatureService->save($creature);
        return $creature;
    }

    private function generateUUID(): string
    {
        $data = random_bytes(16);
        $data[6] = chr(ord($data[6]) & 0x0f | 0x40);
        $data[8] = chr(ord($data[8]) & 0x3f | 0x80);
        return vsprintf('%s%s-%s-%s-%s-%s%s%s', str_split(bin2hex($data), 4));
    }
}
